src/Nethermind/Nethermind.TxPool/NonceTransactionComparerDecorator.cs

```diff
@@ -0,0 +1,49 @@
+﻿//  Copyright (c) 2018 Demerzel Solutions Limited
+//  This file is part of the Nethermind library.
+// 
+//  The Nethermind library is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU Lesser General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+// 
+//  The Nethermind library is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+//  GNU Lesser General Public License for more details.
+// 
+//  You should have received a copy of the GNU Lesser General Public License
+//  along with the Nethermind. If not, see <http://www.gnu.org/licenses/>.
+// 
+
+using System;
+using System.Collections.Generic;
+using Nethermind.Core;
+
+namespace Nethermind.TxPool
+{
+    /// <summary>
+    /// Orders first by <see cref="Transaction.Nonce"/> asc and then by inner comparer
+    /// </summary>
+    public class NonceTransactionComparerDecorator : IComparer<Transaction>
+    {
+        private readonly IComparer<Transaction> _innerComparer;
+
+        public NonceTransactionComparerDecorator(IComparer<Transaction> innerComparer)
+        {
```

Comment:
I think the reason why I still find it confused is that I have thinkf a bit before I understand which of the comparer is apploed first. Maybe it would be better to create a composite comparer that would work from left to right and it would read A ThenBy B ThenBy C and would be generic rather than assumene that nonce comporer has some inner comparers. It seems like a leaking abstraction a bit.